extensions-core/protobuf-extensions/src/main/java/org/apache/druid/data/input/protobuf/ProtobufReader.java

```diff
@@ -88,7 +89,7 @@
   {
     Map<String, Object> record;
 
-    if (flattenSpec == null) {
+    if (flattenSpec == null || JSONPathSpec.DEFAULT.equals(flattenSpec)) {
```

Comment:
is it okay for us to use this path if flattenSpec is set to 
```
{
useFieldDiscovery : true,
fields : []
}
```
because that is what will happen here.

Reply:
So if flattenSpec is set to the default value, in the new flow non root fields are also coming which is expected. Checkout : https://www.javadoc.io/doc/com.google.protobuf/protobuf-java/3.11.0/com/google/protobuf/DynamicMessage.html 
```
Returns a collection of all the fields in this message which are set and their corresponding values.
```

One approach is to document this behavior for proto readers namely non root fields are also part of the row when the above payload is present.

Reply:
I have pushed the null flattenSpec all the way to the `ObjectFlatteners`. This way the behavior of the proto readers will not change.